homeassistant/helpers/config_validation.py

```diff
@@ -267,6 +297,19 @@ def url(value: Any) -> str:
     raise vol.Invalid('invalid url')
 
 
+def email(value: Any) -> str:
+    """Validate an email address."""
+    email_in = str(value)
+    email_re = re.match(
+        r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)",
```

Comment:
wikipedia say:A hostname is considered to be a fully qualified domain name (FQDN) when all labels up to and including the top-level domain name (TLD) are specified. The hostname en.wikipedia.org terminates with the top-level domain org and is thus fully qualified. Depending on the operating system DNS software implementation, an unqualified hostname may be automatically combined with a default domain name configured into the system, in order to complete the fully qualified domain name. As an example, a student at MIT may be able to send mail to "joe@csail" and have it automatically qualified by the mail system to be sent to joe@csail.mit.edu.